instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/datasource/DataSourceInstrumentation.java

```diff
@@ -46,7 +48,8 @@ public static void start(
         return;
       }
 
-      context = instrumenter().start(parentContext, ds);
```

Comment:
Oh, it turns out the javaagent instrumentation was using the library singletons class. Can you add a `dataSourceInstrumenter()` method to the `JdbcSingletons` class? You should be able to reuse the `DataSourceInstrumenterFactory` class you've just added.

Reply:
Are you sure about `JdbcSingletons`? Because It has been renamed to JdbcInstrumenterFactory.

Reply:
Yep - the javaagent instrumentation has its own, separate [`JdbcSingletons`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcSingletons.java) class.